packages/devextreme/js/__internal/data/m_array_query.ts

```diff
@@ -400,6 +391,9 @@ const compileCriteria = (function () {
 
   return function (crit, options) {
     langParams = options || {};
+    const locale = langParams?.locale?.toLowerCase();
+
+    useUpperCase = locale && !!['hy', 'el'].find((code) => locale === code || locale.startsWith(`${code}-`));
```

Comment:
I suggest getting rid of overriding the variable and just declare it here. And then pass it to the functions:

const useUpperCase = locale && !!['hy', 'el'].find((code) => locale === code || locale.startsWith(`${code}-`));
...
compileGroup(crit, useUpperCase);
compileUnary(crit, useUpperCase);
compileBinary(crit, useUpperCase);